server/ingest/http/server.go

```diff
@@ -79,10 +80,23 @@ func (s *Server) putAnnounce(w http.ResponseWriter, r *http.Request) {
 	}
 
 	w.Header().Set("Content-Type", "application/json")
-
 	defer r.Body.Close()
-	err := s.ingestHandler.Announce(r.Body)
-	if err != nil {
+
+	an := message.Message{}
+
+	if r.Header.Get("Content-Type") == "application/json" {
+		if err := json.NewDecoder(r.Body).Decode(&an); err != nil {
+			httpserver.HandleError(w, err, "announce")
+			return
+		}
+	} else {
+		if err := an.UnmarshalCBOR(r.Body); err != nil {
+			httpserver.HandleError(w, err, "announce")
+			return
+		}
+	}
```

Comment:
Remove repeated code:
```suggestion
	var err error
	if r.Header.Get("Content-Type") == "application/json" {
		err = json.NewDecoder(r.Body).Decode(&an)
	} else {
		err = an.UnmarshalCBOR(r.Body)
	}
	if err != nil {
		httpserver.HandleError(w, err, "announce")
		return
	}
```